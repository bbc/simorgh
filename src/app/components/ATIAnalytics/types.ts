import { PageTypes, Platforms, Services } from '../../models/types/global';
import { RequestContextProps } from '../../contexts/RequestContext';
import { ServiceConfig } from '../../models/types/serviceConfig';

export interface AMPAnalyticsData {
  transport: {
    beacon: boolean;
    xhrpost: boolean;
    image: boolean;
  };
  requests: {
    base?: string;
    pageview: string;
  };
  triggers: { trackPageview: { on: string; request: string } };
}

export interface PageData {
  metadata?: {
    analyticsLabels?: {
      counterName?: string;
      contentId?: string;
      // eslint-disable-next-line camelcase
      nations_producer?: string;
      pageIdentifier?: string;
      pageTitle?: string;
      producer?: string;
    };
    atiAnalytics?: {
      producerId?: string;
      chapter?: string;
    };
    locators?: { optimoUrn?: string; curie?: string };
    passport?: {
      category?: { categoryId?: string; categoryName?: string };
      campaigns?: { campaignId?: string; campaignName?: string }[];
      language?: string;
    };
    tags?: {
      about?: {
        thingId?: string;
        thingLabel?: string;
        thingEnglishLabel?: string;
      }[];
    };
    id?: string;
    title?: string;
    language?: string;
    firstPublished?: number;
    lastPublished?: number;
    typ?: PageTypes;
  };
  promo?:
    | { headlines?: { seoHeadline?: string } }
    | { headlines?: { headline?: string } }
    | { subType: string; name: string; uri: string; id: string; type: string };
  id?: string;
  language?: string;
  pageIdentifier?: string;
  pageTitle?: string;
  firstRecordTimeStamp?: string;
  lastRecordTimeStamp?: string;
  contentType?: string;
  title?: string;
}

export interface ATIData {
  campaigns?: { campaignId?: string; campaignName?: string }[] | null;
  categoryName?: string | null;
  contentId?: string | null;
  contentType?: string;
  language?: string | null;
  ldpThingIds?: string | null;
  ldpThingLabels?: string | null;
  nationsProducer?: string | null;
  pageIdentifier?: string;
  pageTitle?: string | null;
  producerId?: string | null;
  producerName?: string | null;
  timePublished?: string | null;
  timeUpdated?: string | null;
}

export interface ATIDataWithContexts {
  requestContext: RequestContextProps;
  serviceContext: ServiceConfig;
  atiData: ATIData;
}

export interface ATIConfigurationDetailsProviders {
  requestContext: RequestContextProps;
  serviceContext: ServiceConfig;
  data?: PageData;
  atiData?: ATIData;
}

export interface ATIAnalyticsProps {
  baseUrl?: string;
  pageviewParams: string;
}

export interface ATIEventTrackingProps {
  campaignID?: string;
  componentName?: string;
  format?: string;
  pageIdentifier?: string;
  platform?: Platforms;
  producerId?: string;
  service?: Services;
  statsDestination?: string;
  type?: string;
  advertiserID?: string;
  url?: string;
}

export interface ATIPageTrackingProps {
  appName?: string;
  contentId?: string | null;
  contentType?: string | null;
  language?: string | null;
  ldpThingIds?: string | null;
  ldpThingLabels?: string | null;
  pageIdentifier?: string;
  pageTitle?: string | null;
  producerId?: string;
  libraryVersion?: string;
  platform?: Platforms;
  statsDestination?: string;
  timePublished?: string | null;
  timeUpdated?: string | null;
  origin?: string;
  previousPath?: string | null;
  categoryName?: string | null;
  campaigns?: { campaignId?: string; campaignName?: string }[] | null;
  nationsProducer?: string | null;
}

export interface ATIProps {
  data?: PageData;
  atiData?: ATIData;
}
