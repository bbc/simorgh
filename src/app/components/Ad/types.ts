import { PageTypes, Services, Direction } from '#app/models/types/global';

export type SlotType = 'leaderboard' | 'mpu';

export interface AdProps {
  slotType: SlotType;
  className?: string;
}

type SlotFunction = (slotType: string) => void;

interface Ads {
  registerSlot: SlotFunction;
  destroySlot: SlotFunction;
}

type PushFunction = (slotFunction: SlotFunction) => void;

interface Cmd {
  push: PushFunction;
}

export interface WindowWithDotCom {
  cmd: Cmd;
  ads: Ads;
}

export interface CanonicalAdBootstrapJsProps {
  adcampaign?: string | null;
}

export interface AdJsonDataProps {
  service: Services;
  slotType: SlotType;
  assetType: string;
}

export interface AmpAdSlotProps {
  service: Services;
  slotType: SlotType;
  pageType: PageTypes;
}

export interface AmpAdContentProps extends AmpAdSlotProps {
  dir: Direction;
  label: string;
}

export interface AmpAdProps extends AmpAdContentProps {
  ariaLabel: string;
}
