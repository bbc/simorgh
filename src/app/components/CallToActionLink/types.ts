import { EventTrackingMetadata } from '#app/models/types/eventTracking';

export type CallToActionLinkProps = {
  href?: string;
  className?: string;
  eventTrackingData?: EventTrackingMetadata;
  download?: boolean;
};
