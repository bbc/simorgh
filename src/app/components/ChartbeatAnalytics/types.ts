import { Services } from '../../models/types/global';
import { MetadataTaggings } from '../../models/types/metadata';

type ChartbeatConfig = {
  domain: string;
  sections: string;
  uid: number;
  title: string;
  virtualReferrer: string | null;
  idSync?: {
    // eslint-disable-next-line camelcase
    bbc_hid: string;
  };
};

export type CanonicalChartbeatConfig = ChartbeatConfig & {
  type?: string;
  useCanonical?: boolean;
};

export interface CanonicalChartbeatProps {
  chartbeatConfig: CanonicalChartbeatConfig;
  chartbeatSource?: string;
}

export type AmpChartbeatConfig = ChartbeatConfig & {
  contentType?: string;
};

export interface AmpChartbeatProps {
  chartbeatConfig: AmpChartbeatConfig;
}

export interface ChartbeatProps {
  sectionName?: string; // required for STY and MAP pages
  categoryName?: string; // required for STY and MAP pages
  mediaPageType?: 'Radio' | 'TV' | 'Podcasts'; // required for Live & On Demand Radio, Podcasts & On Demand TV pages
  title: string;
  taggings?: MetadataTaggings; // required for Media Article Pages
  contentType?: 'player-live' | 'player-episode'; // required for Live & On Demand Radio, Podcasts & On Demand TV pages
  producer?: Services | string; // required for Media Article, STY, MAP, Live & On Demand Radio, Podcasts & On Demand TV pages, if available
  chapter?: string; // required for Media Article, STY, MAP, Live & On Demand Radio, Podcasts & On Demand TV pages, if available
}
