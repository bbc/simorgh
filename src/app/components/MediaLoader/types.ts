import { PageTypes } from '#app/models/types/global';
import { Translations } from '#app/models/types/translations';

export type PlayerConfig = {
  product?: string;
  superResponsive: boolean;
  enableToucan: boolean;
  counterName?: string;
  playlistObject: {
    title: string;
    holdingImageURL: string;
    items: Item[];
  };
  statsObject?: { clipPID?: string };
};

export type Item = {
  versionID: string;
  kind: string;
  duration: number;
  live?: boolean;
};

export type Player = {
  load: () => void;
};

export type BumpType = {
  player: (div: HTMLDivElement | null, config: PlayerConfig) => Player;
};

export type MediaBlock = {
  type: string;
  model: Partial<{
    locator: string;
    text: string;
    title: string;
    blocks: MediaBlock[];
    imageUrl: string;
    versions: {
      versionId: string;
      duration: number;
      warnings?: { [key: string]: string };
    }[];
    smpKind: string;
  }>;
};

export type Props = {
  className: string;
  blocks: MediaBlock[];
};

export type BuildConfigProps = {
  id: string | null;
  pageType: PageTypes;
  blocks: MediaBlock[];
  translations: Translations;
  counterName: string | null;
};
