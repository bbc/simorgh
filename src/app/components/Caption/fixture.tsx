import { v4 as uuid } from 'uuid';
import { blockContainingText } from '#models/blocks';
import { OptimoBlock } from '#app/models/types/optimo';

export default blockContainingText(
  'caption',
  'Some caption text...',
  'mocked-id',
);

export const blockContainingPersianText = blockContainingText(
  'caption',
  'چیسربرگر',
  'mocked-id',
);

export const fragmentBlock = ({
  text,
  attributes = [],
}: {
  text: string;
  attributes?: string[];
}) => ({
  type: 'fragment',
  id: uuid(),
  model: {
    text,
    attributes,
  },
});

export const inlineLinkBlock = ({
  text,
  locator,
  blocks,
  isExternal,
}: {
  text: string;
  locator: string;
  blocks: OptimoBlock[];
  isExternal: boolean;
}) => ({
  type: 'urlLink',
  id: uuid(),
  model: {
    text,
    locator,
    blocks,
    isExternal,
  },
});

export const inlineSpanBlock = ({
  blocks,
  language,
  text,
}: {
  text: string;
  language: string;
  blocks: OptimoBlock[];
}) => ({
  type: 'inline',
  id: uuid,
  model: {
    blocks,
    language,
    text,
  },
});

const persianText = 'چیسربرگر';
const persianLink = inlineLinkBlock({
  text: persianText,
  locator: 'https://google.com',
  blocks: [fragmentBlock({ text: persianText })],
  isExternal: true,
});

const inlinePersianBlock = inlineSpanBlock({
  blocks: [persianLink],
  language: 'fa',
  text: persianText,
});

export const blocksWithInline = {
  model: {
    blocks: [
      {
        model: {
          blocks: [
            {
              model: {
                blocks: [
                  fragmentBlock({
                    text: 'This is some text.',
                    attributes: ['bold'],
                  }),
                  inlinePersianBlock,
                ],
              },
              type: 'text',
            },
          ],
        },
        type: 'caption',
      },
    ],
  },
  type: 'text',
};

export const captionBlock3Paragraphs = {
  model: {
    blocks: [
      {
        model: {
          blocks: [
            {
              model: {
                blocks: [
                  {
                    id: 1,
                    model: {
                      attributes: [],
                      text: 'This is paragraph 1',
                    },
                    type: 'fragment',
                  },
                ],
                text: 'This is paragraph 1',
              },
              type: 'paragraph',
            },
            {
              model: {
                blocks: [
                  {
                    id: 2,
                    model: {
                      attributes: [],
                      text: 'This is paragraph 2',
                    },
                    type: 'fragment',
                  },
                ],
                text: 'This is paragraph 2',
              },
              type: 'paragraph',
            },
            {
              model: {
                blocks: [
                  {
                    id: 3,
                    model: {
                      attributes: [],
                      text: 'Paragraph 3',
                    },
                    type: 'fragment',
                  },
                ],
                text: 'Paragraph 3',
              },
              type: 'paragraph',
            },
          ],
        },
        type: 'text',
      },
    ],
  },
  type: 'caption',
};
