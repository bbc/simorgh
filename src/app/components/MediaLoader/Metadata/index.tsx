import React, { useContext } from 'react';
import { Helmet } from 'react-helmet';
import filterForBlockType from '#app/lib/utilities/blockHandlers';
import { RequestContext } from '#app/contexts/RequestContext';
import {
  ARTICLE_PAGE,
  AV_EMBEDS,
  CPS_ASSET,
  MEDIA_ASSET_PAGE,
  MEDIA_ARTICLE_PAGE,
} from '#app/routes/utils/pageTypes';
import { PageTypes } from '#app/models/types/global';
import { AresMediaBlock, AresMediaMetadataBlock, MediaBlock } from '../types';

const SUPPORTED_PAGE_TYPES = [
  AV_EMBEDS,
  ARTICLE_PAGE,
  CPS_ASSET,
  MEDIA_ASSET_PAGE,
  MEDIA_ARTICLE_PAGE,
] as PageTypes[];

const getThumbnailUri = (url?: string) => {
  if (!url) return null;

  let imageUrl = url;

  if (imageUrl.startsWith('http')) {
    imageUrl = imageUrl.split('/')?.slice(2)?.join('/');
  }

  return `https://${imageUrl.replace('$recipe', '1024x576')}`;
};

const getUploadDate = (availableFrom?: string, firstPublished?: string) => {
  const uploadDate = availableFrom || firstPublished;

  if (!uploadDate) return null;

  return new Date(uploadDate).toISOString();
};

type Props = {
  blocks: MediaBlock[];
  embedURL?: string;
  embedded?: boolean;
};

const Metadata = ({ blocks, embedURL, embedded = false }: Props) => {
  const { pageType } = useContext(RequestContext);

  if (!SUPPORTED_PAGE_TYPES.includes(pageType)) return null;

  const aresMediaBlock: AresMediaBlock = filterForBlockType(
    blocks,
    'aresMedia',
  );

  if (!aresMediaBlock) return null;

  const { model: metadata }: AresMediaMetadataBlock =
    filterForBlockType(aresMediaBlock?.model?.blocks, 'aresMediaMetadata') ??
    {};

  if (!metadata) return null;

  const schema = {
    '@context': 'http://schema.org',
    '@type': metadata?.format === 'audio' ? 'AudioObject' : 'VideoObject',
    name: metadata?.title ?? null,
    description: metadata?.synopses?.short ?? null,
    duration: metadata?.versions?.[0]?.durationISO8601 ?? null,
    thumbnailUrl: getThumbnailUri(metadata?.imageUrl),
    uploadDate: getUploadDate(
      metadata?.versions?.[0]?.availableFrom,
      metadata?.firstPublished,
    ),
    embedURL: embedURL ?? null,
  };

  const metadataJson = Object.fromEntries(
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    Object.entries(schema).filter(([_, v]) => v != null),
  );

  return (
    <Helmet>
      {embedded && embedURL && <meta property="og:url" content={embedURL} />}
      <script type="application/ld+json">{JSON.stringify(metadataJson)}</script>
    </Helmet>
  );
};

export default Metadata;
