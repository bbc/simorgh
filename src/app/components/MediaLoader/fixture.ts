export const aresMediaCaptionBlock = {
  id: '31318aec',
  type: 'caption',
  model: {
    blocks: [
      {
        id: '8ac45ed2',
        type: 'text',
        model: {
          blocks: [
            {
              id: '935f130e',
              type: 'paragraph',
              model: {
                text: 'This is a caption!',
                blocks: [
                  {
                    id: 'deaf6139',
                    type: 'fragment',
                    model: {
                      text: 'This is a caption!',
                      attributes: [],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    ],
  },
};

export const aresMediaPlayerBlock = {
  id: 'c77c0598',
  blockId: 'urn:bbc:ares::clip:p01k6msm',
  type: 'aresMediaMetadata',
  model: {
    id: 'p01k6msm',
    subType: 'clip',
    format: 'video',
    title: 'Five things ants can teach us about management',
    synopses: {
      short:
        'They may be tiny, but us humans could learn a thing or two from ants.',
    },
    imageUrl: 'ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg',
    imageCopyright: 'BBC',
    embedding: true,
    advertising: true,
    versions: [
      {
        versionId: 'p01k6msp',
        types: ['Original'],
        duration: 191,
        durationISO8601: 'PT3M11S',
        warnings: {
          short: 'Contains strong language and adult humour.',
          long: 'Contains strong language and adult humour.',
        },
        availableTerritories: {
          uk: true,
          nonUk: true,
        },
        availableFrom: 1540218932000,
      },
    ],
    syndication: {
      destinations: [],
    },
    smpKind: 'programme',
  },
  position: [5, 2, 1],
};

export const aresMediaBlock = {
  id: '80e150c0',
  type: 'aresMedia',
  model: {
    blocks: [
      aresMediaPlayerBlock,
      {
        id: 'd8f26383',
        type: 'image',
        model: {
          blocks: [
            {
              id: 'fcdba133',
              type: 'rawImage',
              model: {
                width: 1920,
                height: 1080,
                locator:
                  'ichef.test.bbci.co.uk/images/ic/$widthxn/p01k6mtv.jpg',
                originCode: 'mpv',
                copyrightHolder: 'BBC',
              },
            },
            {
              id: '63679c9e',
              type: 'altText',
              model: {
                blocks: [
                  {
                    id: '33876888',
                    type: 'text',
                    model: {
                      blocks: [
                        {
                          id: '26dbfca2',
                          type: 'paragraph',
                          model: {
                            text: 'Ants',
                            blocks: [
                              {
                                id: 'ed9f30c9',
                                type: 'fragment',
                                model: {
                                  text: 'Ants',
                                  attributes: [],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    ],
  },
};

export const livePageCaptionBlock = {
  id: '60db696a',
  type: 'caption',
  model: {
    blocks: [
      {
        id: 'ea8e7d84',
        type: 'text',
        model: {
          blocks: [
            {
              id: 'c7ca4de7',
              type: 'paragraph',
              model: {
                text: 'BBC launch trailer for We Know Our Place women\'s sport campaign"',
                blocks: [
                  {
                    id: 'b33e4302',
                    type: 'fragment',
                    model: {
                      text: 'BBC launch trailer for We Know Our Place women\'s sport campaign"',
                      attributes: [],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    ],
  },
};

export const livePageClipMediaBlock = {
  id: '1ce4d114',
  type: 'clipMedia',
  model: {
    id: 'urn:bbc:pips:pid:p01thw20',
    urns: {
      pipsPid: 'urn:bbc:pips:pid:p01thw20',
    },
    images: [
      {
        url: 'https://ichef.test.bbci.co.uk/images/ic/1024xn/p01thw3g.jpg',
        urlTemplate:
          'https://ichef.test.bbci.co.uk/images/ic/{width}xn/p01thw3g.jpg',
        altText:
          'BBC launch trailer for We Know Our Place women\'s sport campaign"',
        type: 'socialImage',
        source: 'pipsImage',
      },
      {
        url: 'https://ichef.test.bbci.co.uk/images/ic/1024xn/p01thw3g.jpg',
        urlTemplate:
          'https://ichef.test.bbci.co.uk/images/ic/{width}xn/p01thw3g.jpg',
        altText:
          'BBC launch trailer for We Know Our Place women\'s sport campaign"',
        type: 'promoImage',
        source: 'pipsImage',
      },
    ],
    assetPath: 'p01thw20',
    type: 'video',
    headlines: {
      primaryHeadline:
        "BBC launch trailer for We Know Our Place women's sport campaign",
      seoHeadline:
        "BBC launch trailer for We Know Our Place women's sport campaign",
      promoHeadline:
        "BBC launch trailer for We Know Our Place women's sport campaign",
      socialHeadline:
        "BBC launch trailer for We Know Our Place women's sport campaign",
    },
    analytics: {
      page: {
        name: 'programmes.av.p01thw20.page',
        contentId: 'urn:bbc:pips:pid:p01thw20',
        producer: 'PROGRAMMES',
      },
    },
    description:
      'BBC launch trailer for We Know Our Place women\'s sport campaign"',
    summary: {
      type: 'text',
      model: {
        blocks: [
          {
            type: 'paragraph',
            model: {
              text: 'BBC launch trailer for We Know Our Place women\'s sport campaign"',
              blocks: [
                {
                  type: 'fragment',
                  model: {
                    text: 'BBC launch trailer for We Know Our Place women\'s sport campaign"',
                    attributes: [],
                  },
                },
              ],
            },
          },
        ],
      },
    },
    lastPublished: '2022-07-01T08:56:56Z',
    firstPublished: null,
    video: {
      id: 'p01thw20',
      title: "BBC launch trailer for We Know Our Place women's sport campaign",
      holdingImage: {
        id: 'https://ichef.test.bbci.co.uk/images/ic/$recipe/p01thw3g.jpg',
        altText:
          'BBC launch trailer for We Know Our Place women\'s sport campaign"',
      },
      version: {
        id: 'p01thw22',
        duration: 'PT54S',
        kind: 'programme',
        guidance: null,
        territories: ['nonuk', 'uk'],
      },
      isAdvertisingAllowed: true,
      isEmbeddingAllowed: true,
      isUnavailable: false,
    },
    attributions: null,
    link: {
      path: '/programmes/p01thw20',
    },
    section: null,
    isSharingAllowed: true,
  },
};

export const onDemandTvBlock = {
  type: 'tv',
  model: {
    id: 'w172zm8b4tlpzxh',
    subType: 'episode',
    format: 'Video',
    title: '06/09/2024 GMT',
    synopses: {
      short:
        'ताज़ा अंतरराष्ट्रीय, क्षेत्रीय ख़बरों और विश्लेषण के लिए देखिए बीबीसी दुनिया',
      medium:
        'ताज़ा अंतरराष्ट्रीय, क्षेत्रीय ख़बरों और विश्लेषण के लिए देखिए बीबीसी दुनिया',
    },
    imageUrl: 'ichef.bbci.co.uk/images/ic/$recipe/p0hfjjfk.png',
    embedding: false,
    advertising: false,
    versions: [
      {
        versionId: 'w1mskypb14t285q',
        types: ['Original'],
        duration: 1192,
        durationISO8601: 'PT19M52S',
        warnings: {},
        availableTerritories: {
          uk: true,
          nonUk: true,
          world: false,
        },
        availableFrom: 1725641390000,
        availabilityStatus: 'available',
      },
    ],
    availability: 'available',
    smpKind: 'programme',
    episodeTitle: 'दुनिया',
    type: 'media',
  },
};

export const mediaOverridesBlock = {
  model: {
    language: 'hi',
    pageIdentifierOverride: 'hindi.bbc_hindi_tv.tv.w172zm8b4tlpzxh.page',
    pageTitleOverride: 'दुनिया',
  },
  type: 'mediaOverrides',
};

export const aresMediaLiveStreamBlocks = [
  {
    id: '3e29a5f5',
    type: 'aresMedia',
    model: {
      blocks: [
        {
          id: 'd67fb17d',
          type: 'aresMediaMetadata',
          blockId: 'urn:bbc:ares::primary:108540166',
          model: {
            live: true,
            embedding: false,
            subType: 'primary',
            id: '108540166',
            available: true,
            format: 'audio_video',
            title: 'مباشر: تلفزيون بي بي سي عربي',
            imageCopyright: 'BBC',
            imageUrl:
              'http://c.files.bbci.co.uk/CF4E/production/_111607035_arabic_16_9_updated.png',
            synopses: {
              short: 'مباشر: تلفزيون بي بي سي عربي',
              medium: 'مباشر: تلفزيون بي بي سي عربي',
              long: 'مباشر: تلفزيون بي بي سي عربي',
            },
            versions: [
              {
                kind: 'programme',
                live: true,
                versionId: 'bbc_arabic_tv',
              },
            ],
          },
        },
      ],
    },
  },
  aresMediaCaptionBlock,
];

export const aresMediaBlocks = [aresMediaBlock, aresMediaCaptionBlock];
export const clipMediaBlocks = [livePageClipMediaBlock, livePageCaptionBlock];
export const onDemandTvBlocks = [onDemandTvBlock];
export const onDemandTvBlocksWithOverrides = [
  onDemandTvBlock,
  mediaOverridesBlock,
];
