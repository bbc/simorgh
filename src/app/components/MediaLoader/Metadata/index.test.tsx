import React from 'react';
import { Helmet } from 'react-helmet';
import {
  act,
  render,
} from '#app/components/react-testing-library-with-providers';
import Metadata from '.';
import { aresMediaBlocks } from '../fixture';
import { MediaBlock } from '../types';

describe('Media Loader - Metadata', () => {
  const embedSource =
    'https://www.test.bbc.com/ws/av-embeds/articles/c3wmq4d1y3wo/p01k6mtv/en-GB';

  it('should not render Metadata when aresMedia has no metadata block ', async () => {
    let container;

    await act(async () => {
      ({ container } = render(<Metadata blocks={[]} embedURL={embedSource} />));
    });

    expect(container).toBeEmptyDOMElement();
  });

  it('should render Metadata correctly when aresMedia has a metadata block', async () => {
    await act(async () => {
      render(
        <Metadata
          blocks={aresMediaBlocks as MediaBlock[]}
          embedURL={embedSource}
        />,
      );
    });

    const container = Helmet.peek();
    expect(container).toMatchSnapshot();
  });
});
