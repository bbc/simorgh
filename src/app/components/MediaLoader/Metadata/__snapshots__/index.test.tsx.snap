// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Loader - Metadata should render Metadata correctly when aresMedia has a metadata block 1`] = `
{
  "baseTag": [],
  "bodyAttributes": {},
  "defer": true,
  "encode": true,
  "htmlAttributes": {},
  "linkTags": [],
  "metaTags": [],
  "noscriptTags": [],
  "onChangeClientState": [Function],
  "scriptTags": [
    {
      "innerHTML": "{"@context":"http://schema.org","@type":"VideoObject","name":"Five things ants can teach us about management","description":"They may be tiny, but us humans could learn a thing or two from ants.","duration":"PT3M11S","thumbnailUrl":"https://ichef.test.bbci.co.uk/images/ic/1024x576/p01k6mtv.jpg","uploadDate":"2018-10-22T14:35:32.000Z","embedURL":""}",
      "type": "application/ld+json",
    },
  ],
  "styleTags": [],
  "title": undefined,
  "titleAttributes": {},
}
`;
